src/components/modals/ModalBody.js

```diff
@@ -2,16 +2,42 @@
  * @flow
  */
 
-import type { ComponentType } from 'react';
+import React, { useEffect } from 'react';
 
+import _isFunction from 'lodash/isFunction';
 import styled from 'styled-components';
 
-const ModalBody :ComponentType<{|
-  children ?:any;
-  width ?:string;
-|}> = styled.div`
+const ModalBodyWrapper = styled.div`
   max-width: 100%;
   width: ${({ width }) => ((typeof width === 'number' && width > 0) ? width : 720)}px;
 `;
 
+const ModalBody = ({
+  children,
+  onCleanUp,
+  width,
+} :{|
+  children :any;
+  onCleanUp ?:() => void;
+  width ?:number;
+|}) => {
+
+  useEffect(() => () => {
+    if (_isFunction(onCleanUp)) {
+      onCleanUp();
+    }
+  }, [onCleanUp]);
```

Comment:
I don't think this abstraction is necessary and is actually detrimental. Reasons will be detailed in later comments.

Reply:
I think this is worth keeping around.